pyramid/viewderivers.py

```diff
@@ -464,48 +460,38 @@ def decorated_view(view, info):
 
 decorated_view.options = ('decorator',)
 
-def _parse_csrf_setting(val, error_source):
-    if val:
-        if isinstance(val, string_types):
-            if val.lower() in truthy:
-                val = True
-            elif val.lower() in falsey:
-                val = False
-        elif not isinstance(val, bool):
-            raise ConfigurationError(
-                '{0} must be a string or boolean value'
-                .format(error_source))
-    return val
-
-SAFE_REQUEST_METHODS = frozenset(["GET", "HEAD", "OPTIONS", "TRACE"])
-
 def csrf_view(view, info):
-    default_val = _parse_csrf_setting(
-        info.settings.get('pyramid.require_default_csrf'),
-        'Config setting "pyramid.require_default_csrf"')
-    explicit_val = _parse_csrf_setting(
-        info.options.get('require_csrf'),
-        'View option "require_csrf"')
-    resolved_val = explicit_val
-    if (explicit_val is True and default_val) or explicit_val is None:
-        resolved_val = default_val
-    if resolved_val is True:
-        resolved_val = 'csrf_token'
+    explicit_val = info.options.get('require_csrf')
+    defaults = info.registry.queryUtility(IDefaultCSRFOptions)
```

Comment:
Wouldn't it be simpler to always register a default utility, that can be overridden, rather than having to redefine the defaults here?

If we ever decide to change the safe_methods or the header, we need to change it in two places.